src/akkudoktoreos/core/ems.py

```diff
@@ -113,36 +172,94 @@ def __init__(
             if isinstance(parameters.einspeiseverguetung_euro_pro_wh, list)
             else np.full(len(self.gesamtlast), parameters.einspeiseverguetung_euro_pro_wh, float)
         )
+        if wechselrichter is not None:
+            self.akku = wechselrichter.akku
+        else:
+            self.akku = None
         self.eauto = eauto
         self.home_appliance = home_appliance
         self.wechselrichter = wechselrichter
-        self.ac_charge_hours = np.full(config.prediction_hours, 0)
-        self.dc_charge_hours = np.full(config.prediction_hours, 1)
-        self.ev_charge_hours = np.full(config.prediction_hours, 0)
+        self.ac_charge_hours = np.full(self.config.prediction_hours, 0.0)
+        self.dc_charge_hours = np.full(self.config.prediction_hours, 1.0)
+        self.ev_charge_hours = np.full(self.config.prediction_hours, 0.0)
 
-    def set_akku_discharge_hours(self, ds: np.ndarray) -> None:
-        self.akku.set_discharge_per_hour(ds)
+    def set_akku_discharge_hours(self, ds: List[int]) -> None:
+        if self.akku is not None:
+            self.akku.set_discharge_per_hour(np.array(ds, int))
 
     def set_akku_ac_charge_hours(self, ds: np.ndarray) -> None:
         self.ac_charge_hours = ds
 
     def set_akku_dc_charge_hours(self, ds: np.ndarray) -> None:
         self.dc_charge_hours = ds
 
-    def set_ev_charge_hours(self, ds: np.ndarray) -> None:
+    def set_ev_charge_hours(self, ds: List[float]) -> None:
         self.ev_charge_hours = ds
```

Comment:
performance problem (4% total time single_test) because pydantic validation, also function input should stay ndarray (change one call in genetic)